### PR TITLE
Implementation of token auth for all endpoints.

### PR DESCRIPTION
**Pull request by LyonesGamer**
Replaces password and ldap authentication on all endpoints with token auth, adds a /login endpoint which uses password or ldap authentication to generate a short-term token for use on other endpoints. Completes #201.

### CODE REVIEW

**Comment by ElijahCaine** [comment]
After a pretty thorough review I'd say this all looks good.
mfw the diff is ~7k lines
Once the diff gets resolve and we get a few more eyes on this it should be good to go.

**Comment by ElijahCaine** [comment]
lgtm :+1:

File: src/login.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
+'use strict';
+const errors = require('./errors');
+const passport = require('passport');
+const crypto = require('crypto');
+
+const tokens = {};
+
+const MAX_AGE = 30 * 60 * 1000;
```
Comment (Kennric) [inline src/login.js]: This should, at some point, be a configurable number

**Comment by Kennric** [comment]
I can't see any issues, please create a new issue to make token age configurable. +1

**Pull request merged**